### DISCUSSION TITLE
CORS Missing Allow origin when migrating to 9.8

### DISCUSSION THREAD

**Discussion by victorsmits** [directus/directus#12623] [Archive]
I was running on an old version of docker directus for a while and after a small fix on the app the directus backend upgrade itself to the latest version. However now I cannot gather data from the backend ( via de JDK) the error message I receive is a CORS error. A weird thing is that the query has no issue when running in postman. Can some one help me

**Comment by rijkvanzanten** [comment]
See https://docs.directus.io/self-hosted/config-options.html#cors 🙂

**Reply by victorsmits** [reply]
I have try but nothing work
this is my current config env
![Capture d’écran 2022-04-07 à 22 31 24](https://user-images.githubusercontent.com/22278315/162290792-95f94758-3f9e-4980-8258-31d357d9474a.png)

**Reply by rijkvanzanten** [reply]
Ohh in the above screenshot, you're quite a couple versions behind (`9.0.0` instead of `9.8.0`) 🤔 

Also... you just shared a couple passwords in a public repo, so you might want to rotate out the passwords / keys / secrets 😬

**Reply by victorsmits** [reply]
ho yes wrong screenshot sorry it was a try to recover my old working version

**Reply by victorsmits** [reply]
Here is the error I got with the latest version

![Capture d’écran 2022-04-07 à 22 41 50](https://user-images.githubusercontent.com/22278315/162294077-87d6e421-70d0-4dc0-ad50-1de687736247.png)

**Reply by victorsmits** [reply]
here is the response from the option call

![Capture d’écran 2022-04-07 à 22 43 00](https://user-images.githubusercontent.com/22278315/162294813-f2fbcc36-df0a-4f20-9cfc-f44d66252a78.png)

**Reply by rijkvanzanten** [reply]
Are you sure that's the response from the options call? The options call in the first screenshot also throws a cord error about the missing credentials header, but it's included (and status 204) in your second screenshot

**Reply by victorsmits** [reply]
I finally managed to fix the api, I had to add each domain calling the api by hand. It turns out that the * does not work

**Reply by rijkvanzanten** [reply]
Yup, `*` will set the header to `*`, which browsers don't necessarily agree with. As per the docs, you can use `true` to dynamically match the incoming origin, or provide a CSV of domains 👍🏻 

> Value for the Access-Control-Allow-Origin header. Use true to match the Origin header, or provide a domain or a CSV of domains for specific access

https://docs.directus.io/self-hosted/config-options.html#cors

**Reply by kieusonlam** [reply]
> I finally managed to fix the api, I had to add each domain calling the api by hand. It turns out that the * does not work

Can I see your docker-compose file pls, i'm stuck on this too.

**Reply by guillem-gelabert** [reply]
I fixed it by adding
```
      CORS_ENABLED: "true"
      CORS_ORIGIN: "true"
```

**Comment by boooong123** [comment]
> I fixed it by adding
> 
> ```
>       CORS_ENABLED: "true"
>       CORS_ORIGIN: "true"
> ```

Correct me if i'm wrong but I think "CORS_ORIGIN: true" will allow any request from any domain to be accepted?

**Comment by linin-nin** [comment]
I also cors block when I try to CreateItem. what should I do?
here is my function:

 const handleSubmit = async(e:any) => {
    e.preventDefault();
    await client.request(
      createItem('Subscripber', {
        email: Email,
        phone_number: Phonenum
      })
    )
    
    console.log('success')
  }

**Reply by kostyaostapuk** [reply]
@rijkvanzanten also for me

**Reply by rijkvanzanten** [reply]
This is a years old discussion about a version that's long deprecated. If you have an issue on the latest version, please open a new issue
